### PR TITLE
Imb 51  Are you submitting late? Why are you submitting late? 

### PR DESCRIPTION
**Pull request by muthrajan**
## What
-  Add screens for Are you submitting late? Why are you submitting late?  with validation [IMB-51](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-51)

## Why
- Screen and logic to match the flow in Figma, as part of the IMA MVP

## How
- Add html files for the content to match the Figma screens
- Add relevant path name for the new screen in index.js
- Add relevant fields in /fields/index.js
- Add validation for text area, date field, CSS for date legend
- Use can upload files for late submission details

## Testing
- Tests passing locally

### CODE REVIEW

**Review by Rhodine-orleans-lindsay — CHANGES_REQUESTED** [review]
Please see comments. Can you also make your commit more specific according to the recommended format i.e.

JIRA NUMBER ticket title

bullet point 1
bullet point 2, etc

Have a look at the commit here as an example: https://github.com/UKHomeOffice/ima/pull/50

File: apps/ima/translations/src/en/pages.json [hunk 1]
```diff
@@ -24,6 +24,15 @@
   "evidence-upload": {
     "header": "Upload evidence"
   },
+  "are-you-submitting-this-form-late": {
+    "header": "Are you submitting this form late?",
+    "paragraph1": "Select 'no' if there is no end to the claim period on your duty to remove notice, and you have not been given another notice with an end date on it",
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json:29]: paragraph1 removed and added it to 'are-you-submitting-this-form-late' markdown.

File: apps/ima/views/are-you-submitting-this-form-late.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{{<partials-page}} {{$page-content}} {{#markdown}}are-you-submitting-this-form-late{{/markdown}} <br><br>
+  <p class="govuk-body">{{#t}}pages.are-you-submitting-this-form-late.paragraph1{{/t}}</p>
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/are-you-submitting-this-form-late.html:2]: paragraph1 removed in html  and added  to the 'are-you-submitting-this-form-late' markdown

File: apps/ima/views/are-you-submitting-this-form-late.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{{<partials-page}} {{$page-content}} {{#markdown}}are-you-submitting-this-form-late{{/markdown}} <br><br>
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/are-you-submitting-this-form-late.html:1]: Formatting done

File: apps/ima/translations/src/en/pages.json [hunk 4]
```diff
@@ -24,6 +24,15 @@
...
+    "header1": "Have you asked for an extension?"
+  }, 
+  "why-are-you-submitting-late": {
+    "header": "Why are you submitting this form late?",
+    "paragraph1": "You must upload evidence of why you could not submit in time"
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json:34]: paragraph1 removed added to markdown

File: apps/ima/views/partials/late-submission-extension-details.html [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<fieldset id="late-submission-extension-details-fieldset" class="govuk-radios__conditional" aria-hidden="false">
+{{#renderField}}are-you-submitting-this-form-late-extension-options-yes-detail{{/renderField}}
+ </fieldset>
+
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/partials/late-submission-extension-details.html:4]: new line corrected

File: apps/ima/index.js [hunk 6]
```diff
@@ -45,6 +45,23 @@ module.exports = {
...
+      fields: [
+        'are-you-submitting-this-form-late',
+        'are-you-submitting-this-form-late-extension',
+        'are-you-submitting-this-form-late-extension-options-yes-detail'
+      ],
+      continueOnEdit: true,
+      next: '/why-are-you-submitting-late'
+    },
+    '/why-are-you-submitting-late': {
+      behaviours: [SaveImage('image'), RemoveImage, LimitDocument],
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/index.js:62]: locals: { showSaveAndExit: true },  added to index.js

File: apps/ima/views/are-you-submitting-this-form-late.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  <p class="govuk-body">{{#t}}pages.are-you-submitting-this-form-late.paragraph1{{/t}}</p>
+  {{#renderField}}are-you-submitting-this-form-late{{/renderField}}
+  <br><br>
+  {{#renderField}}are-you-submitting-this-form-late-extension{{/renderField}}
+  <br><br>
+  {{#input-submit}}continue{{/input-submit}}
+  {{/page-content}}
+  {{/partials-page}}
+
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/are-you-submitting-this-form-late.html:10]: new line corrected

File: apps/ima/views/are-you-submitting-this-form-late.html [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
   {{#renderField}}are-you-submitting-this-form-late{{/renderField}}
   <br><br>
   {{#renderField}}are-you-submitting-this-form-late-extension{{/renderField}}
   <br><br>
   {{#input-submit}}continue{{/input-submit}}
   {{/page-content}}
   {{/partials-page}}
-  
\ No newline at end of file
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/are-you-submitting-this-form-late.html:10]: newline

File: apps/ima/views/declaration.html [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{{<partials-page}} {{$page-content}}
+ {{#markdown}}declaration{{/markdown}}
+ <br><br>
+  {{#renderField}}person-declaration{{/renderField}} 
+  <br><br>
+  <br><br>
+  <p>You will not be able to change your answers after you submit.</p>
+  <br><br>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/declaration.html:8]: Why are there so many` <br>`?

File: apps/ima/views/helper-declaration.html [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+ {{#markdown}}helper-declaration{{/markdown}}
+ <br><br>
+  {{#renderField}}helper-declaration{{/renderField}} 
+  <br><br>
+  <p>You will not be able to change your answers after you submit.</p>
+  <br><br>
+  {{#input-submit}}confirm-submission{{/input-submit}}
+  {{/page-content}}
+  {{/partials-page}}
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/helper-declaration.html:15]: new line

File: apps/ima/views/immigration-adviser-declaration.html [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  {{#renderField}}use-interpreter{{/renderField}}
+ {{#markdown}}immigration-adviser-declaration{{/markdown}}
+ <br><br>
+ {{#renderField}}immigration-adviser-declaration{{/renderField}} 
+  <p>You will not be able to change your answers after you submit.</p>
+  <br><br>
+  {{#input-submit}}confirm-submission{{/input-submit}}
+  {{/page-content}}
+  {{/partials-page}}
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/immigration-adviser-declaration.html:14]: new line

File: apps/ima/translations/src/en/pages.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rhodine-orleans-lindsay) [reply apps/ima/translations/src/en/pages.json:34]: Please remove it from this file as well not just the .html

File: apps/ima/translations/src/en/pages.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rhodine-orleans-lindsay) [reply apps/ima/translations/src/en/pages.json:29]: Please remove it from this file as well not just the .html

[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json:29]: paragraph1 removed from  pages.json

File: apps/ima/translations/src/en/pages.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json:34]: Yes it is removed from pages.json file

File: apps/ima/views/are-you-submitting-this-form-late.html (hunk 8 above)
Comment (muthrajan) [reply apps/ima/views/are-you-submitting-this-form-late.html:10]: New line corrected

File: apps/ima/translations/src/en/fields.json [hunk 12]
```diff
@@ -18,6 +18,35 @@
       }
     }
   },
+  "are-you-submitting-this-form-late": {
+    "legend": " ",
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/fields.json:22]: Legend amended

File: apps/ima/translations/src/en/pages.json [hunk 13]
```diff
@@ -24,6 +24,15 @@
   "evidence-upload": {
     "header": "Upload evidence"
   },
+  "are-you-submitting-this-form-late": {
+    "header": "Are you submitting this form late?",
+    "paragraph1": "Select 'no' if there is no end to the claim period on your duty to remove notice, and you have not been given another notice with an end date on it",
+    "header1": "Have you asked for an extension?"
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json]: header1 removed

File: apps/ima/views/immigration-adviser-declaration.html (hunk 11 above)
Comment (muthrajan) [reply apps/ima/views/immigration-adviser-declaration.html]: New line added

File: apps/ima/views/helper-declaration.html (hunk 10 above)
Comment (muthrajan) [reply apps/ima/views/helper-declaration.html:15]: New line added

File: apps/ima/views/declaration.html (hunk 9 above)
Comment (muthrajan) [reply apps/ima/views/declaration.html:9]: <br> removed

File: apps/ima/translations/src/en/pages.json [hunk 14]
```diff
@@ -63,29 +60,31 @@
...
-    "paragraph3": "You can email evidence to [EMAIL ADDRESS]",
-    "paragraph4": "Your email must include the Unique application number [UAN] and date of birth you used to sign in to this form.",
-    "paragraph5": "Information about how to send more evidence is also on the email we send you when you submitted this form.",
+    "paragraph3": "If you have given a claim period and need to send evidence after the end of the period, you need to:",
+    "paragraph4": "1. apply for an extension at the end of your claim period",
+    "paragraph5": "2. send your evidence to [EMAIL ADDRESS] as soon as possible",
     "paragraph6": "Do not wait for confirmation that you have been granted an extension to send your evidence.",
     "paragraph7": "Your email subject line must contain the Unique application number {{values.uan}} and date of birth you used to sign in to this form.",
     "paragraph8": "Information about how to send more evidence is also on the email we send you when you submitted this form."
-
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/translations/src/en/pages.json]: I'd put this in a markdown

File: apps/ima/translations/src/en/pages.json [hunk 15]
```diff
@@ -63,29 +60,31 @@
     "header": "Declaration"
   },
   "immigration-adviser-declaration": {
+    "adviser": "Immigration adviser declaration",
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/translations/src/en/pages.json:63]: I don't think this is needed here, the title is already showing as 'Declaration'

File: apps/ima/views/summary.html [hunk 16]
```diff
@@ -1,3 +1,4 @@
+<title>{{#t}}pages.summary.summary{{/t}} – GOV.UK</title>
 {{<partials-page}}
   {{$page-content}}
     <table class="table-details">
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/summary.html]: I think you should remove this file completely and use the confirm.html file instead. You can add the template to the step in the ima/index.js like this:

> behaviours: [Summary, SaveFormSession],
> sections: require('./sections/summary-data-sections'),
> locals: { showSaveAndExit: true },
> template: 'confirm',

This is because the summary page at the end of the form will have to change it's path name from '/summary' to '/final-summary' and the summary page at the beginning of the form is already using a summary html file which has different information in it.

File: apps/ima/views/immigration-adviser-declaration.html [hunk 17]
```diff
@@ -1,7 +1,7 @@
+<title>{{#t}}pages.immigration-adviser-declaration.adviser{{/t}} – GOV.UK</title>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/immigration-adviser-declaration.html:1]: I don't think this is needed here, the title is already showing as 'Declaration'

File: apps/ima/views/helper-declaration.html [hunk 18]
```diff
@@ -1,7 +1,7 @@
+<title>{{#t}}pages.helper-declaration.helper{{/t}} – GOV.UK</title>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/helper-declaration.html:1]: I don't think this is needed here, the title is already showing as 'Declaration'

File: apps/ima/views/declaration.html [hunk 19]
```diff
@@ -1,3 +1,4 @@
+<title>{{#t}}pages.declaration.header{{/t}} – GOV.UK</title>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/declaration.html:1]: I don't think this is needed here, should pull automatically from pages.json

File: apps/ima/views/confirmation.html [hunk 20]
```diff
@@ -19,8 +20,12 @@ <h2 class="govuk-heading-m">{{#t}}pages.confirmation.header2{{/t}}</h2>
   <p class="govuk-body">{{#t}}pages.confirmation.paragraph2{{/t}}</p>
 
   <h3 class="govuk-heading-m">{{#t}}pages.confirmation.header3{{/t}}</h2>
+    <p class="govuk-body">{{#t}}pages.confirmation.paragraph3{{/t}}</p>
     <p class="govuk-body">{{#t}}pages.confirmation.paragraph4{{/t}}</p>
     <p class="govuk-body">{{#t}}pages.confirmation.paragraph5{{/t}}</p>
+    <p class="govuk-body">{{#t}}pages.confirmation.paragraph6{{/t}}</p>
+    <p class="govuk-body">{{#t}}pages.confirmation.paragraph7{{/t}}</p>
+    <p class="govuk-body">{{#t}}pages.confirmation.paragraph8{{/t}}</p>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/confirmation.html]: I'd put this in a markdown

File: apps/ima/views/are-you-submitting-this-form-late.html [hunk 21]
```diff
@@ -1,3 +1,4 @@
+<title>{{#t}}pages.are-you-submitting-this-form-late.header{{/t}} – GOV.UK</title>
 {{<partials-page}} {{$page-content}}
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/are-you-submitting-this-form-late.html:2]: Put  {{$page-content}}  on the next line

File: apps/ima/views/declaration.html [hunk 22]
```diff
@@ -1,3 +1,4 @@
...
 {{<partials-page}} {{$page-content}}
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/declaration.html:2]: put {{$page-content}} on the next line

File: apps/ima/views/confirm.html [hunk 23]
```diff
@@ -1,3 +1,4 @@
+<title>{{#t}}pages.confirm.header{{/t}} – GOV.UK</title>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/confirm.html:1]: remove this - it is already being pulled from hof framework

File: apps/ima/views/confirm.html [hunk 24]
```diff
@@ -1,3 +1,4 @@
...
 {{<partials-page}}
   {{$page-content}}
     <table class="table-details">
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/confirm.html:4]: Add `<h1 class="gov-uk-heading-l">{{#t}}pages.confirm.header{{/t}}</h1>` above the table

File: apps/ima/views/are-you-submitting-this-form-late.html [hunk 25]
```diff
@@ -7,4 +7,3 @@
   {{#input-submit}}continue{{/input-submit}}
   {{/page-content}}
   {{/partials-page}}
-
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/are-you-submitting-this-form-late.html]: New line corrected

File: apps/ima/views/immigration-adviser-declaration.html [hunk 26]
```diff
@@ -11,4 +11,3 @@
   {{#input-submit}}confirm-submission{{/input-submit}}
   {{/page-content}}
   {{/partials-page}}
-
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/immigration-adviser-declaration.html]: New line corrected

File: apps/ima/translations/src/en/pages.json [hunk 27]
```diff
@@ -63,29 +60,31 @@
...
+    "helper": "Helper declaration",
     "header": "Declaration",
     "paragraph1": "What language was used to read the completed form to {{values.name}}?",
-    "paragraph2": "Include any dialects",
-    "header1": "Did you use an interpreter?"
+    "paragraph2": "Include any dialects"
+  },
+  "summary": {
+    "summary": "Summary"
   },
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json]: Summary removed

File: apps/ima/translations/src/en/pages.json (hunk 14 above)
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json]: New markdown created and applied css style.

File: apps/ima/translations/src/en/pages.json [hunk 28]
```diff
@@ -63,29 +60,31 @@
...
   "immigration-adviser-declaration": {
+    "adviser": "Immigration adviser declaration",
     "header": "Declaration",
     "paragraph1": "What language was used to read the completed form to {{values.name}}?",
-    "paragraph2": "Include any dialects",
-    "header1": "Did you use an interpreter?"
+    "paragraph2": "Include any dialects"
   },
   "helper-declaration": {
+    "helper": "Helper declaration",
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json]: Removed

File: apps/ima/views/confirmation.html (hunk 20 above)
Comment (muthrajan) [reply apps/ima/views/confirmation.html]: We have now form-submitted.html with markdown. This file is removed.

File: apps/ima/views/are-you-submitting-this-form-late.html (hunk 21 above)
Comment (muthrajan) [reply apps/ima/views/are-you-submitting-this-form-late.html]: {{$page-content}}  moved to new line

File: apps/ima/views/declaration.html (hunk 22 above)
Comment (muthrajan) [reply apps/ima/views/declaration.html]: {{$page-content}} moved to a new line.

File: apps/ima/views/confirm.html (hunk 24 above)
Comment (muthrajan) [reply apps/ima/views/confirm.html:5]: <h1 class="gov-uk-heading-l">{{#t}}pages.confirm.header{{/t}}</h1>  added

File: apps/ima/translations/src/en/pages.json (hunk 15 above)
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json:63]: Removed

File: apps/ima/views/summary.html (hunk 16 above)
Comment (muthrajan) [reply apps/ima/views/summary.html]: Removed

File: apps/ima/views/immigration-adviser-declaration.html (hunk 17 above)
Comment (muthrajan) [reply apps/ima/views/immigration-adviser-declaration.html:1]: Removed

File: apps/ima/views/helper-declaration.html (hunk 18 above)
Comment (muthrajan) [reply apps/ima/views/helper-declaration.html:1]: Removed

File: apps/ima/views/declaration.html (hunk 19 above)
Comment (Rhodine-orleans-lindsay) [reply apps/ima/views/declaration.html:1]: As discussed, this is not needed

File: apps/ima/views/are-you-submitting-this-form-late.html [hunk 29]
```diff
@@ -1,11 +1,11 @@
-<title>{{#t}}pages.are-you-submitting-this-form-late.header{{/t}} – GOV.UK</title>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/are-you-submitting-this-form-late.html]: Why was this removed?

File: apps/ima/views/declaration.html (hunk 19 above)
Comment (Rhodine-orleans-lindsay) [reply apps/ima/views/declaration.html:1]: As discussed, this is not needed

File: apps/ima/views/immigration-adviser-declaration.html [hunk 30]
```diff
@@ -1,14 +1,14 @@
-<title>{{#t}}pages.immigration-adviser-declaration.adviser{{/t}} – GOV.UK</title>
+<!-- <title>{{#t}}pages.immigration-adviser-declaration.adviser{{/t}} – GOV.UK</title> -->
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/immigration-adviser-declaration.html:1]: Remove this

File: apps/ima/views/confirm.html [hunk 31]
```diff
@@ -1,7 +1,7 @@
 <title>{{#t}}pages.confirm.header{{/t}}</title>
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/confirm.html:1]: missing the `- GOV.UK` in the title

File: apps/ima/translations/src/en/pages.json (hunk 15 above)
Comment (Rhodine-orleans-lindsay) [reply apps/ima/translations/src/en/pages.json:63]: This has not been removed

File: apps/ima/translations/src/en/pages.json (hunk 14 above)
Comment (Rhodine-orleans-lindsay) [reply apps/ima/translations/src/en/pages.json]: Once you have put this in a markdown it should be removed from the pages.json

File: apps/ima/views/content/form-submitted.md [hunk 32]
```diff
@@ -0,0 +1,12 @@
+**If you need to send more evidence**
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/views/content/form-submitted.md]: make this `### If you need to send more evidence`

File: apps/ima/translations/src/en/pages.json (hunk 15 above)
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json]: Now removed

File: apps/ima/translations/src/en/pages.json (hunk 14 above)
Comment (muthrajan) [reply apps/ima/translations/src/en/pages.json]: header3 added to markdown  with ### and removed from html.

**Review by Rhodine-orleans-lindsay — APPROVED** [review]

File: apps/ima/index.js [hunk 33]
```diff
@@ -33,16 +38,90 @@ module.exports = {
...
-      next: '/confirm', // TO BE UPDATED AS STEPS ARE ADDED
+      next: '/evidence-upload', // TO BE UPDATED AS STEPS ARE ADDED
       backLink: 'current-progress'
     },
-    '/confirm': {
+    '/evidence-upload': {
+      behaviours: [SaveImage('image'), RemoveImage, LimitDocument],
+      fields: ['image'],
+      continueOnEdit: true,
+      next: '/are-you-submitting-this-form-late'
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/index.js:48]: this should be the summary page

File: apps/ima/index.js [hunk 34]
```diff
@@ -33,16 +38,90 @@ module.exports = {
...
+          }
+        },
+        {
+          target: '/helper-declaration',
+          condition: {
+            field: 'who-are-you',
+            value: 'helper'
+          }
+        }
+      ]
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/index.js:95]: this should be part of the why-are-you-submitting -late step

File: apps/ima/index.js [hunk 35]
```diff
@@ -33,16 +38,90 @@ module.exports = {
...
+      locals: { showSaveAndExit: true },
+      fields: ['image', 'late-submission'],
+      continueOnEdit: true,
+      next: '/summary'
+    },
+    '/summary': {
       behaviours: [Summary, SaveFormSession],
       sections: require('./sections/summary-data-sections'),
       locals: { showSaveAndExit: true },
-      next: '/confirmation'
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/index.js:43]: this should be '/are-you-submitting-this-form-late'

**Review by Rhodine-orleans-lindsay — CHANGES_REQUESTED** [review]
Please rearrange the steps in ima/index.js according to the figma

**Review by Rhodine-orleans-lindsay — APPROVED** [review]

File: apps/ima/index.js (hunk 33 above)
Comment (muthrajan) [reply apps/ima/index.js]: changes to summary page

File: apps/ima/index.js (hunk 34 above)
Comment (muthrajan) [reply apps/ima/index.js]: Changed to part of why-are-you-submitting-late stamp

File: apps/ima/index.js (hunk 35 above)
Comment (muthrajan) [reply apps/ima/index.js:43]: Changed to submitting-late

**Review by mislam987 — APPROVED** [review]
Latest changes look good to me.
